source/Projectile.cpp

```diff
@@ -115,7 +111,14 @@ void Projectile::Move(vector<Visual> &visuals, vector<Projectile> &projectiles)
 
 			for(const auto &it : weapon->Submunitions())
 				for(size_t i = 0; i < it.count; ++i)
-					projectiles.emplace_back(*this, it.offset, it.facing, it.weapon);
+				{
+					// Determine the inaccuracy to add to a new projectile
+					Angle projectileInaccuracy;
+					double outfitInaccuracy = weapon->Inaccuracy();
```

Comment:
This should be hoisted out of the loop.

Probably a free function or static class method should be provided that provides a single source of truth for converting from an inaccuracy value to the sampled angle, now that the generation is happening in more than one source file.

Reply:
I'm still not sure what you're referring to when you say free method.  I know how to make a method that doesn't have any class scope within one of the .cpp files, but then I don't think anybody outside the .cpp file can access it.  I could probably make a library class though, might take me a bit.

Reply:
This [StackOverflow](https://stackoverflow.com/a/4863328/9337071) answer explains the term.

Since we do have a decent context for the usage of the function (an inaccuracy angle is only relevant when creating a projectile), it should be a static method of the Projectile class, e.g.

```c++
class Projectile : ... {
public:
  static Angle computeInaccuracy(double value);

public:
  ... Other stuff
}
```

The body would be responsible for checking the input against 0.

Reply:
I saw that article but there was a bit of a disconnect for me on what was part of a class (and still a little bit).  Seems to me what you're looking for is a public, static function (so we can use it from any reference, specifically with the submunitions that only have the strict at that point) stored in the same file as the related class (projectile).  I can't do that until tonight but I'll definitely get on it when I can.  It'll also provide a neat little spot for uniform inaccuracy to be toggled between when it seems like the right time to add it.

Reply:
Just pushed this improvement